packages/app-builder/src/components/ui/user/link.tsx

```diff
@@ -0,0 +1,82 @@
+import { useEditor } from '@craftjs/core'
+import React from 'react'
+import { Link as RRLink } from 'react-router-dom'
+import { getPages } from '../header/saveState'
+import { ToolbarItem, ToolbarItemType, ToolbarSection } from '../toolbar'
+import Container, { ContainerProps } from './container'
+
+const defaultProps = {
+  destination: '/',
+}
+
+interface LinkProps extends ContainerProps {
+  destination?: string
+  context?: { [key: string]: any }
+}
+
+const qs = (query: { [key: string]: string }) => {
+  return Object.keys(query)
+    .map((key) => `${key}=${query[key]}`)
+    .join('&')
+}
+
+const Link = (props: LinkProps) => {
+  const { enabled } = useEditor((state) => ({
+    enabled: state.options.enabled,
+  }))
+
+  return (
+    <RRLink
+      to={{
+        pathname: props.destination,
+        search: props.context ? qs(props.context) : '',
+      }}
+      onClick={
+        enabled
+          ? (e) => {
+              e.preventDefault()
+              return false
+            }
+          : undefined
```

Comment:
What's this for?

Reply:
so that when it's in edit mode and you click the link to edit the text contents it doesn't take you to the destination page

Reply:
ahhh I see!